test/services/NewAuthYesNoSubmissionServiceSpec.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import base.{AppWithDefaultMockFixtures, SpecBase}
+import models.UserAnswers
+import models.requests.DataRequest
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.*
+import pages.NewAuthYesNoPage
+import play.api.mvc.Results.Redirect
+import play.api.mvc.{AnyContent, Call, Results}
+import utils.transformers.IE043Transformer
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Success
+
+class NewAuthYesNoSubmissionServiceSpec extends SpecBase with AppWithDefaultMockFixtures {
+
+  private val mockTransformer               = mock[IE043Transformer]
+  private val service                       = new NewAuthYesNoSubmissionService(mockTransformer)
+  implicit private val ec: ExecutionContext = ExecutionContext.global
+
+  val userAnswers        = mock[UserAnswers]
+  val transformedAnswers = mock[UserAnswers]
```

Comment:
Mocking a case class, while possible, is a little unusual. Could these tests be rewritten such that we're only mocking the dependencies (in this case the IE043Transformer)? See the json in the [old controller test](https://github.com/hmrc/manage-transit-movements-unloading-frontend/blob/c979d5fc6f6d1621ef485c414cb57317b8272040/test/controllers/NewAuthYesNoControllerSpec.scala#L115-L190) as an example.